lisa/features/resize.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from lisa.feature import Feature
+
+if TYPE_CHECKING:
+    from lisa.sut_orchestrator.azure.platform_ import AzureCapability
+
+FEATURE_NAME_RESIZE = "Resize"
+
+
+class ResizeAction(str, Enum):
+    IncreaseCoreCount = "IncreaseCpuCount"
+    DecreaseCoreCount = "DecreaseCpuCount"
+
+
+class Resize(Feature):
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_RESIZE
+
+    @classmethod
+    def can_disable(cls) -> bool:
+        raise NotImplementedError()
+
+    def enabled(self) -> bool:
+        raise NotImplementedError()
+
+    def is_supported(self) -> bool:
+        raise NotImplementedError()
+
+    def resize(self, resize_action: ResizeAction = None) -> "AzureCapability":
```

Comment:
It doesn't need to return the AzureCapability, return the `NodeSpace`.

Reply:
done